src/tactics.ml

```diff
@@ -114,3 +114,19 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
       handle_refine (Rewrite.reflexivity ps)
   | P_tac_sym           ->
       handle_refine (Rewrite.symmetry ps)
+  | P_tac_why3(s)       ->
+      (* get the goal to prove *)
+      let (_, trm) = Proof.Goal.get_type g in
+      let trm = unfold trm in
+      let prover_name = Why3prover.get_name s.elt in
+      let proved = Why3prop.t_goal prover_name trm in
+      if proved then
+        let why3_axiom = Pos.none "Why3Axiom" in
+        let current_sign = ss.signature in
+        (* TODO : add the goal to the set of axioms if the prover succeed *)
+        let a = Sign.add_symbol current_sign Const why3_axiom trm in
+        Console.out 2 "%s proved the current goal@." prover_name;
+        handle_refine (Symb(a, Nothing))
+      else
```

Comment:
Handling of local assumptions is missing.

Reply:
Fixed in 0dfaee931f2933120d308aa19635dfb42f117985